### PR TITLE
[cpp] Use auto-casting rules with RawPointer

### LINKED ISSUE #889: Pointer casting doesn't work with RawPointer<T> but does with Star<T>
When casting between pointers, the correct output is generated if `Star<T>` is used but not if `RawPointer<T>` is used:

```haxe
import cpp.*;

var buffer: Star<cpp.Void>;
var floatBuffer: Star<Float32> = cast buffer;
// generates correct: float* floatBuffer = ( (float*) buffer ) 🎉

var buffer: RawPointer<cpp.Void>;
var floatBuffer: RawPointer<Float32> = cast buffer;
// generates incorrect: float* floatBuffer = buffer
// the lack of native casting means the compiler throws an error here
```

I think this is a `gencpp.ml` issue

---
\* I've opened a few issues without PRs recently which I feel like brings negativity but I want to add that I absolutely _love_ hxcpp and hope to be using it a lot more in the future! These issues help me keep track of things I want to make PRs for this year

### PR DESCRIPTION
**Pull request by haxiomic**
See https://github.com/HaxeFoundation/hxcpp/issues/889

Casting with raw pointer doesn't generate the correct output
```haxe
import cpp.*;

var buffer: RawPointer<cpp.Void>;
var floatBuffer: RawPointer<Float32> = cast buffer;
// generates incorrect: float* floatBuffer = buffer
// the lack of native casting means the compiler throws an error here

var buffer: Star<cpp.Void>;
var floatBuffer: Star<Float32> = cast buffer;
// generates correct: float* floatBuffer = ( (float*) buffer ) 🎉
```

Adding `TCppRawPointer` to the list types to use the auto-casting rules (like `TCppPointer` and `TCppStar` fixes it)

I'm not very familiar with gencpp.ml so it would be great if someone who was could check this for me – is it likely that RawPointer is also missing from other places?

closes https://github.com/HaxeFoundation/hxcpp/issues/889

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
Can you add a test, which would fail in development branch, but pass in this PR?

**Comment by haxiomic** [comment]
@RealyUniqueName test added, fails on development, passes in this PR

File: tests/unit/src/unit/hxcpp_issues/Issue9194.hx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package unit.hxcpp_issues;
+
+
+class Issue9194 extends Test {
+
+	@:analyzer(no_optimize)
+	function test() {
+		#if (cpp && !cppia)
```
Comment (RealyUniqueName) [inline tests/unit/src/unit/hxcpp_issues/Issue9194.hx]: Why the test is disabled for cppia?

Comment (haxiomic) [reply tests/unit/src/unit/hxcpp_issues/Issue9194.hx]: Mainly because I've never tried cppia :P, let me give it a try

**Comment by haxiomic** [comment]
I think this one should be good* to go now 👍

**Pull request merged**